torchvision/models/swin_transformer.py

```diff
@@ -39,18 +39,23 @@ def __init__(self, dim: int, norm_layer: Callable[..., nn.Module] = nn.LayerNorm
         self.norm = norm_layer(4 * dim)
 
     def forward(self, x: Tensor):
-        B, H, W, C = x.shape
-
-        x0 = x[:, 0::2, 0::2, :]  # B H/2 W/2 C
-        x1 = x[:, 1::2, 0::2, :]  # B H/2 W/2 C
-        x2 = x[:, 0::2, 1::2, :]  # B H/2 W/2 C
-        x3 = x[:, 1::2, 1::2, :]  # B H/2 W/2 C
-        x = torch.cat([x0, x1, x2, x3], -1)  # B H/2 W/2 4*C
-        x = x.view(B, -1, 4 * C)  # B H/2*W/2 4*C
```

Comment:
I have confirmed that:
```
x = x.view(B, -1, 4 * C)
x = self.norm(x)
x = self.reduction(x)
x = x.view(B, H // 2, W // 2, 2 * C)
```
is equivalent with
```
 x = self.norm(x)
x = self.reduction(x)
```
using `nn.LayerNorm` as the `norm`

Reply:
> @xiaohu2015 I wonder if you have the bandwidth to check the proposed updates in this PR.
> 
> To provide some context, Yosua is making some small adjustments to the original implementation so that we can support easier the 3D version of the Swin model. Please let us know if you have any thoughts.

I think the adjustments don't affect current model, but can @jdsgomes double check the model accuary?

Reply:
I don't think the variation is due to this. This is more likely due to the non-determinism. If you run it a few times, you will get different results.